jax/_src/dispatch.py

```diff
@@ -414,7 +416,8 @@ def raise_warnings_or_errors_for_jit_of_pmap(nreps, backend, name, jaxpr):
 @profiler.annotate_function
 def lower_xla_callable(
     fun: lu.WrappedFun, device, backend, name, donated_invars,
-    always_lower: bool, keep_unused: bool, *arg_specs):
+    always_lower: bool, keep_unused: bool, *arg_specs,
+    lowering_platform: Optional[str]):
```

Comment:
experimental_lowering_platform

Reply:
Actually, I call it "experimental" only in the `lower()` API, which is public. The rest are internal APIs, and I think it mostly adds clutter.

Reply:
I think it adds clarification for us (jax devs) as to what it is supposed to be. So I think experimental is a good choice for variables too

Reply:
Maybe a comment can be a happy medium? For a jax dev audience, I'm less concerned at this point with labeling it experimental, and more concerned with it not being obvious what it is supposed to mean.

Reply:
The problem is that there are so many codepaths currently that you would have to add a comment everywhere for it to be obvious what it means.

This code in dispatch.py is dead code (lol) and a lot of other places this CL changes. It will be deleted after March 15. That's why if the variable name also contains experimental, then it is obvious what is going on.

Reply:
Anyways, not a big deal just wanted to see if it was reasonable. 

If you feel strongly, then lowering_platform is fine.

Reply:
Adding `experimental` everywhere seems to only increase code size (`experimental_platform_lowering=experimental_platform_lowering` is the most common use). I would prefer that we do it only at the API surface.

Adding a lot of comments for these one-line uses is also not very useful, imo. The key usage is at the API surface, when I do not think we should add comments, or where it actually gets used: `lowering_platform or backend.platform`. I could add comments in those 3 places, but at that point it is very obvious what is going on. wdyt?

Reply:
Ok, sounds good. Adding a comment in pxla.py near the call to mlir.<something> should be good enough I think.

The dispatch.py codepath is dead code and will be removed in 2 weeks anyways.

Reply:
Done